Controllers/UsersController.cs

```diff
@@ -134,6 +136,18 @@ public IActionResult AccessDenied()
         return Forbid();
     }
 
+    [HttpGet("contractors")]
+    [AllowAnonymous]
+    public async Task<IActionResult> GetAllContractors(){
+
+        var contractors = await _userManager.GetUsersInRoleAsync("Contractor");
+        var contractorsInfo = contractors.Select(user => new{
```

Comment:
Just in case: there is UserDetailsDTO.cs with properties username and email, yoy may (bout do not have to) find it useful